src/modules/poolModule.ts

```diff
@@ -497,19 +498,19 @@ export class PoolModule implements IModule {
     )
 
     const args = [
-      tx.pure(globalPauseStatusObjectId),
-      tx.pure(poolsId),
-      tx.pure(params.tick_spacing.toString()),
-      tx.pure(params.initialize_sqrt_price),
-      tx.pure(params.uri),
+      tx.object(globalPauseStatusObjectId),
+      tx.object(poolsId),
+      tx.pure.u32(params.tick_spacing),
+      tx.pure.u128(params.initialize_sqrt_price),
+      tx.pure.string(params.uri),
       primaryCoinAInputsR.targetCoin,
       primaryCoinBInputsR.targetCoin,
-      tx.pure(asUintN(BigInt(params.tick_lower)).toString()),
-      tx.pure(asUintN(BigInt(params.tick_upper)).toString()),
```

Comment:
I am not sure about the formatting of `asUintN`.   I know u32 is the right pure encoding, but not sure if params.tick_loweer/upper are in the right format, I expect this might be wrong, and should be reverted to something like:  `tx.pure.u32(Number(asUintN(BigInt(params.tick_lower))))`